src/components/NumberModals/EmergencyContactTable.jsx

```diff
@@ -0,0 +1,105 @@
+import {
+  Tbody,
+  Th,
+  Tr,
+  Td,
+  Menu,
+  MenuList,
+  MenuItem,
+  IconButton,
+  MenuButton,
+  useDisclosure,
+} from '@chakra-ui/react';
+import { BsThreeDotsVertical } from 'react-icons/bs';
+
+import { OCHBackend } from '../../common/utils';
+import CommonTable from '../../common/CommonTable/CommonTable';
+import { CommonTableHeader } from '../../common/CommonTable/CommonTableHeader';
+import DeleteNumberModal from './DeleteNumberModal';
+import EditNumberModal from './EditNumberModal';
+
+const EmergencyContactTable = ({ tableData, setTableData }) => {
+  const [numberId, setNumberId] = useState(-1);
+  const [rowData, setRowData] = useState({});
+  const editModalDisclosure = useDisclosure();
+  const deleteModalDisclosure = useDisclosure();
+
+  const openModalWithData = (dataRow, openFunc) => {
+    setNumberId(dataRow._id);
+    setRowData(dataRow);
+    openFunc();
+  };
+
+  const editNumber = async newNumber => {
+    await OCHBackend.put(`/numbers/${numberId}`, {
+      name: newNumber.name,
+      number: newNumber.number,
+      note: newNumber.note,
+    });
+    const res = await OCHBackend.get('/numbers');
+    setTableData(res.data);
+  };
+
+  const deleteNumber = async () => {
+    await OCHBackend.delete(`/numbers/${numberId}`);
+    const res = await OCHBackend.get('/numbers');
+    setTableData(res.data);
+  };
+
+  return (
+    <>
+      <EditNumberModal
+        numberData={rowData}
+        editNumber={editNumber}
+        disclosure={editModalDisclosure}
+      />
+      <DeleteNumberModal deleteNumber={deleteNumber} disclosure={deleteModalDisclosure} />
+
+      <CommonTable>
+        <CommonTableHeader>
+          <Th fontWeight="500">Contact Name</Th>
+          <Th>Number</Th>
+          <Th> </Th>
+          <Th> </Th>
+        </CommonTableHeader>
+        <Tbody>
+          {tableData.map(row => (
+            <Tr key={row._id}>
+              <Td fontWeight="500">{row.name}</Td>
+              <Td>{row.number}</Td>
+              <Td>{row.note}</Td>
+              <Td>
+                <Menu isLazy autoSelect={false}>
+                  <MenuButton>
+                    <IconButton icon={<BsThreeDotsVertical />} bg="transparent" />
+                  </MenuButton>
+                  <MenuList>
+                    <MenuItem onClick={() => openModalWithData(row, editModalDisclosure.onOpen)}>
+                      Edit Contact
+                    </MenuItem>
+                    <MenuItem
+                      color="red.600"
+                      onClick={() => openModalWithData(row, deleteModalDisclosure.onOpen)}
+                    >
+                      Delete Contact
+                    </MenuItem>
+                  </MenuList>
+                </Menu>
+              </Td>
+            </Tr>
+          ))}
+        </Tbody>
+      </CommonTable>
+    </>
+  );
+};
+
+EmergencyContactTable.propTypes = {
+  tableData: PropTypes.arrayOf(PropTypes.object).isRequired,
```

Comment:
🚫 **[eslint]** <[react/forbid-prop-types](https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/forbid-prop-types.md)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Prop type "object" is forbidden